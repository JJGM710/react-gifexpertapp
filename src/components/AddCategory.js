import React, { useState } from 'react'
import PropTypes from 'prop-types';

const AddCategory = ({setCategories, categories}) => {

    const [inputValue,setInputValue]= useState('')

    const handleInputChange = (e) => {
        //console.log(e.target.value);
        setInputValue(e.target.value)


        console.log('llamdo de evento onchange')
    }

    const handleSubmit = (e) =>{
        e.preventDefault();
        //console.log('handleSubmit', inputValue)
        let test = categories.find((ctg) => ctg.toLowerCase() === inputValue.trim().toLowerCase());
        if (inputValue.trim().length > 2 && test === undefined)
        {
            setCategories(ctg => [inputValue, ...ctg]);
            console.log('se hizo el cambio')
            setInputValue('');
        }
       // console.log(categories);
        
    }
    return (
        <form onSubmit={handleSubmit}>
            <p> {inputValue} </p>
            <input 
                placeholder="Search.."
                className='box'
                type = "text" 
                value = {inputValue}
                onChange = {handleInputChange}
            ></input>
        </form>
    )
}

AddCategory.propTypes = {
    setCategories : PropTypes.func.isRequired,
    categories: PropTypes.array.isRequired
};

export default AddCategory;
