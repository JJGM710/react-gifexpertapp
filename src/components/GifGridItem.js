import React from 'react'
import { PropTypes } from 'prop-types';

export const GifGridItem = ({id, title,url}) => {
   // console.log(id, title, url);
    return (
        <div className="card animate__animated animate__fadeIn animate__delay-1s">
            <img src={url} alt={title} />
            <p> {title?title.toLowerCase():'titulo no definido'} </p>
        </div>
    )
}


//definicon de default props y de isrequired


//validacion de propiedades
GifGridItem.propTypes = {
    title : PropTypes.string.isRequired,
    url : PropTypes.string.isRequired
};

//definir propiedades por defect
GifGridItem.defaultProps = {
};


// 1.enzyme
// 2.enzyme to jSON
// 3. debe mostar el comp correctamente
    
// shallow
// wrapper 
// expect wrapper toMatchSnapshot 
