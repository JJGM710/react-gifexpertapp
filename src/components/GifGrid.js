import React from 'react'
import { GifGridItem } from './GifGridItem';
//import { getGifs } from './../helpers/getGifs';
import { useFetchGifs } from './../hooks/useFetchGifs';

export const GifGrid = ({category}) => {

    // const [images, setimages] = useState([]);

    // // genera una dependecia con category, es decir, que cada vez que el valor de esa propiedad cambie va a renderizar ese component
    // useEffect(() => {
    //     getGifs(category)
    //         .then(imgs => setimages(imgs));
    // }, [category])

    const {data : images, loading} = useFetchGifs(category);


    //getGifs();
    return (
        <>
        <h3 className="animate__animated animate__flash animate__slow">{category}</h3>

        {loading && <h2 className="animate__animated animate__flash animate__slower">Loading...</h2>}

        <div className= 'card-grid'>
            
            {images.map((img) => (
                 <GifGridItem 
                    key = {img.id}
                    {...img} />
            ))
            }

        </div>
        </>
    );
};
