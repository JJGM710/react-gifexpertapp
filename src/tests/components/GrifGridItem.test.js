import  React from 'react';
import { GifGridItem } from '../../components/GifGridItem';
import { shallow } from 'enzyme';
import '@testing-library/jest-dom';

const category = ['One Punch Man']

const title = 'titulo';
const url = 'https://localhost/algo.jpg';
const id = '112334';

const wrapper = shallow(<GifGridItem title = {title} url = {url} id = {id}/>);

describe('Pruebas en GifGridItem', () => {
    test('debe renderizar correctamente', () => {

        expect(wrapper).toMatchSnapshot();
    })

    test('debe decir un parrafo con el title... revisar el valor del title ', () => {
        const p = wrapper.find('p');
        expect(p.text().trim()).toBe(title);
        
    })

    test('debe decir una imagen... revisar url y alt de los props', () => {
        const img = wrapper.find('img');
        console.log(img.props().src);
        expect(img.props().src).toBe(url);
        expect(img.props().alt).toBe(title);
    })
    
    test('debe tener animate__fadeIn ', () => {
        //metodo dos para comparar class names
        // const div = wrapper.find('div').hasClass('animate__fadeIn');
        // console.log(div)
        // expect(div).toBe(true)

        const div = wrapper.find('div');
        const className = div.prop('className');

        expect(className.includes('animate__fadeIn')).toBe(true);
        
    })
    
    
    
    
})
