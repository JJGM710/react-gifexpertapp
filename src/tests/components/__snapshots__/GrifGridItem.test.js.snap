// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en GifGridItem debe renderizar correctamente 1`] = `
<div
  className="card animate__animated animate__fadeIn animate__delay-1s"
>
  <img
    alt="titulo"
    src="https://localhost/algo.jpg"
  />
  <p>
     
    titulo
     
  </p>
</div>
`;
